ssh.go

```diff
@@ -68,29 +70,68 @@ func (s *SSHServer) handleTunnelCommand(sess ssh.Session) {
 	}
 }
 
-func (s *SSHServer) handleSessionWithCommand(sess ssh.Session) {
-	cmd := sess.Command()[0]
-	s.logger.Debugw("received command", "command", cmd)
+func (s *SSHServer) fillCommandsMap(commandsMap map[string]int, commands []string) {
 
-	cmdKey, cmdValue := s.parseCommand(cmd)
-	switch cmdKey {
-	case "tunnel":
-		shouldTunnel, err := strconv.ParseBool(cmdValue)
-		if err != nil {
-			s.logger.Errorw("failed to parse tunnel command", "error", err)
-			return
-		}
-		if !shouldTunnel {
-			s.handleBasicSession(sess)
-			return
+	commandsMap["ttl"] = MaxTtl //default value
+
+	for i := 0; i < len(commands); i++ {
+		cmd := commands[i]
+
+		s.logger.Debugw("received command", "command", cmd)
+		cmdKey, cmdValue := s.parseCommand(cmd)
+
+		switch cmdKey {
+		case "tunnel": //if user wants to confirm tunnel
+			shouldTunnel, err := strconv.ParseBool(cmdValue)
+			if err != nil {
+				s.logger.Errorw("failed to parse tunnel command", "error", err)
+				return
+			}
+
+			var bit = 0
+			if shouldTunnel {
+				bit = 1
+			}
+			commandsMap["tunnel"] = bit
+
+		case "ttl": //if user wants to specify ttl time in seconds
+			ttl, err := strconv.ParseInt(cmdValue, 10, 32)
+
+			if err != nil {
+				s.logger.Errorw("failed to parse ttl command", "error", err)
+				return
+			}
+
+			if ttl > MaxTtl {
+				ttl = MaxTtl
+			}
+			if ttl <= MinTtl {
+				ttl = MinTtl
+			}
+
+			commandsMap["ttl"] = int(ttl)
+
+		default:
+			s.logger.Warnw("unknown command", "command", cmd)
 		}
-		s.handleTunnelCommand(sess)
-	default:
-		s.logger.Warnw("unknown command", "command", cmd)
+
 	}
+
 }
 
-func (s *SSHServer) handleBasicSession(sess ssh.Session) {
+func (s *SSHServer) handleSessionWithCommand(sess ssh.Session) {
+
+	commandsMap := make(map[string]int)
```

Comment:
why use an `int` as the value type for the `map`? You know that `map[string]struct{}` is more memory efficient (even though in this case it doesn't matter, since the map is never going to be big), and more go idiomatic for lookups